### PR TITLE
feat(frontend): enable the XRP Ledger integration

### PR DESCRIPTION
**Pull request by sbpublic**
# Motivation

Phase 10 (formerly 9) of the XRP Ledger integration: stop force-disabling XRP, so it behaves exactly like BTC/ETH/SOL. Stacked on #13593 → #13595, and depends on the backend prerequisite #13596.

XRP has shipped disabled behind a temporary in-code override for the whole stack. Removing it is what makes the chain real, and it is the first time the suite-wide network/token expectations have to know about XRP.

# Changes

- **Enablement**: delete `XRP_MAINNET_DISABLED_OVERRIDE`, leaving the standard `VITE_XRP_MAINNET_DISABLED` convention (enabled by default) on every environment including the vitest suite.
- **Endpoint handling made standard**: `VITE_XRP_RPC_URL_MAINNET` moves from `env/networks/networks.xrp.env.ts` to `env/rest/xrpl.env.ts`, where every other provider credential lives, and the `xrplcluster.com` fallback is dropped. A build without the variable now has no XRPL endpoint and `xrpHttpRpcUrl` throws, exactly as a build without an Alchemy or QuickNode key fails its requests; the fallback made an unconfigured build look like a working one. The variable keeps its name and value shape, so no workflow or secret changes, and `docs/ai/integrations/xrpl.md` now lists the three ways to point it at a public cluster deliberately — a local `.env`, the staging secret, or the `env-override` dispatch input — since removing the fallback does not put those endpoints out of reach. It still holds a whole URL where the others hold a key composed onto a hostname in code — splitting it that way means committing the XRPL hostname and re-issuing the secrets as bare tokens, deliberately left for later.
- **Network settings**: map XRP mainnet to and from the new `NetworkSettingsFor::XrpMainnet` key (`networkIdToKey` / `keyToNetworkId`), so a user's enable/disable choice for XRP is persisted like every other network. Without it, `networkIdToKey` warned `Unknown networkId: XRP` and — because `keyToNetworkId` ends in an exhaustive `assertNever` — the file no longer type-checked once the backend variant existed.
- **Suite-wide expectations**: XRP now appears in the enabled network/token lists (`networks.derived`, `tokens.derived`, `all-tokens.derived`, `all-network-tokens.derived`, `network-tokens.derived`). It sorts directly after Solana among the native tokens (see Ordering below).
- **Price**: XRP gets a USD price on the default frontend path, which fetches straight from CoinGecko — `exchangeRateXRPToUsd` under the `ripple` coin id (XRP's CoinGecko id is not `xrp`), `currentXrpPrice` threaded through the exchange worker and the provider fallback, and the store's `ripple` entry mapped onto `XRP_TOKEN_ID`. The optional backend exchange path also resolves it, via `XRP_NATIVE_ENTRY` (`TokenId::XrpNativeMainnet`), so it does not need a fallback outcall; `tokenIdKey` learns the new variant.
- **Price fallback guard**: `fillXrp` is now part of the exchange worker's nothing-missing predicate, so the CoinGecko fallback still runs when XRP is the only native the backend omitted — previously that case returned early and XRP stayed unpriced.
- **Ordering**: native XRP sits directly after Solana rather than trailing the EVM tokens, and `XRP_TOKEN` is pinned after `SOLANA_TOKEN` in `tokensToPin`, so it is treated like every other native network. The token list order is ICP, BTC, ETH, SOL, XRP, then EVM.
- **Docs**: an XRP Ledger section in `PRODUCT.md` (destination tags; the whole account reserve — base plus one owner reserve per owned ledger object, which is what `getXrpReserveDrops` subtracts and what the fee context shows), XRP added to the supported-asset list, and `docs/ai/integrations/xrpl.md` corrected — it named `VITE_XRP_MAINNET_ENABLED`, a flag that never existed, and still described `fee` / `submit` / `account_tx` as future work. The opening description now says "threshold cryptography" rather than "threshold ECDSA", which is wrong for the SOL and XRP it lists — both sign with threshold Schnorr over Ed25519.
- **Activity specs**: seed the XRP transactions store alongside the other chains' in `AllTransactionsList`, `AllTransactions` and `AllTransactionsSkeletons`, since XRP now joins the activity view's loading check and an uninitialized store otherwise pinned the view to skeletons.
- **Spec**: the rollout section moves to the past tense with the shipped state stated after it — the override is gone and `VITE_XRP_MAINNET_DISABLED` alone governs XRP — and its "no CI/deploy env plumbing" claim is bounded to enablement, since the RPC endpoint always needed the staging/beta secrets and a production value from the build. Three comments that cited `.env.example` as the everyday source of an empty endpoint no longer do, now that it ships a working one. Phases renumbered — 9 is the backend prerequisite off `main`, 10 is this PR.

# Divergence from the spec

- `breaking-interface` is **expected to be red here** and is deliberately not addressed: the gate fires on any PR whose diff touches `src/backend/backend.did`, which this one only does via the merged #13596 commits. Marking a frontend PR as an API break would be wrong; the check clears once #13596 lands on `main` and the backend diff leaves this PR — which has to happen before this can merge anyway.
- This branch carries a **merge of #13596** so the XRP backend variants and their generated declarations are available. Those backend commits drop out of the diff once #13596 merges to `main`.
- An earlier revision of this PR gated `XRP_MAINNET_ENABLED` on the endpoint being present, justified by "`deploy-to-environment.yml` has no `ic` arm for it". That reasoning was wrong: the workflow has no `ic` target at all — its choices are staging, beta, `test_fe_1-6`, `test_be_1` and audit — and no variable has an `ic` arm, Alchemy and QuickNode included. Production is not deployed from this repo's workflows, so XRP's wiring there is as complete as every other provider's. The gate is reverted; XRP is enabled by `VITE_XRP_MAINNET_DISABLED` alone, like every other chain.
- **Activity gate settles on a failed first load**: `syncWalletError` now writes the XRP transactions entry when nothing ever loaded, so a token whose first load failed no longer holds `areTransactionsStoresLoaded` open — which kept an otherwise-empty Activity view on skeletons and stopped `levelNewcomers` running. The same write covers the omitted-history path, where an `account_tx` failure is folded into `undefined` and posted as a normal wallet update, so `syncWalletError` never runs and the gate was held open silently while balances kept succeeding. The write is conditional, unlike the other chains' unconditional `reset`/`nullify`: an entry that already holds rows keeps them. Known cost: once settled, the aggregate view reports XRP as having no activity rather than spinning — better than a permanent skeleton, but surfacing an explicit history-failure state is a separate change.
- **`.env.example` gets a working endpoint**: `VITE_XRP_RPC_URL_MAINNET=https://xrplcluster.com`. With XRP enabled and no `VITE_XRP_MAINNET_DISABLED` line in that file, a developer copying the example would otherwise get XRP as the one enabled non-IC chain, with no endpoint and every request throwing.

# Tests

- New/updated price tests: XRP is queried under `ripple` (`exchange.services`), maps onto `XRP_TOKEN_ID` (`exchange.derived`), and is threaded through the worker on both the direct and backend paths (`exchange.worker`, 74 passing). `XrpLoaderWallets` tests were rewritten — they asserted the disabled premise, which enabling inverts. All 118 `tests/xrp` tests pass.
- Updated the five suite-wide network/token expectation specs; `networks.derived`, `user-networks.utils`, `user-networks.derived`, `tokens.derived`, `all-tokens.derived`, `all-network-tokens.derived` and `network-tokens.derived` all pass.
- Ran locally: `npm run check` and `npm run check:tests` — **0 errors, 0 warnings** — plus `prettier --check` and `eslint --max-warnings 0` on the changed files, and a full sweep of `tests/lib/derived` + `tests/lib/utils` (3434 passed).
- The 26 remaining local failures in that sweep are all the known `localStorage is not a function` jsdom artifact (`active-user-transactions.derived`, `theme.utils`, `privacy.utils` — they fail in `beforeEach`, unrelated to this change). Full suite + coverage gate run on CI.
- Review follow-ups: four new `xrp-listener.services` cases — the aggregate gate closing after a failed first load (asserted through the real `areTransactionsStoresLoaded`), and a loaded entry surviving an error. Two on the balance-failure path and two on the omitted-history path, each pairing the gate closing (asserted through the real `areTransactionsStoresLoaded`) with loaded rows surviving. All verified by drift: removing either write fails its gate case, writing unconditionally fails its rows-survive case. One new `exchange.worker` case omits only XRP from the backend response, so the early return is what is under test (verified by drift — removing `!fillXrp` fails it). The endpoint spec moves to `tests/env/rest/xrpl.env.spec.ts` and loses its `PROD`/`BETA` cases, which only existed to pin which flavours got the fallback; it now covers a configured endpoint, `TEST`, an empty value, and an absent variable resolving to no endpoint at all.

## Not verified

The enable path has **not** been exercised against a real deploy, and the send path has never run against a real ledger — there is no XRP testnet in the integration to dry-run against (it is mainnet-only). Separately, a build with no `VITE_XRP_RPC_URL_MAINNET` now has no XRPL endpoint at all — the `xrplcluster.com` fallback is gone — so XRP is visible and its requests fail, the same way ETH behaves without an Alchemy key. The staging and beta secrets are provisioned; local development needs the variable set.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code) — model: Claude Opus 5 (claude-opus-5)

### CODE REVIEW

**Review by AntonioVentilii — APPROVED** [review]

File: src/frontend/src/env/networks/networks.xrp.env.ts [hunk 1]
```diff
@@ -8,23 +8,13 @@ import { defineSupportedNetworks } from '$lib/utils/env.networks.utils';
...
-// in the final "enable" PR. Remove this override (and the `!… &&` below) in that PR —
-// XRP then behaves exactly like BTC/ETH/SOL.
-const XRP_MAINNET_DISABLED_OVERRIDE = PROD || BETA || TEST;
-
-export const XRP_MAINNET_ENABLED =
-	!XRP_MAINNET_DISABLED_OVERRIDE &&
-	parseEnabledMainnetBoolEnvVar(import.meta.env.VITE_XRP_MAINNET_DISABLED);
+export const XRP_MAINNET_ENABLED = parseEnabledMainnetBoolEnvVar(
+	import.meta.env.VITE_XRP_MAINNET_DISABLED
+);
```
[the remark this answers is not in the document]
Comment (sbpublic) [reply src/frontend/src/env/networks/networks.xrp.env.ts:15]: Correct, and the wiring is thinner than the finding assumes.

**Summary**: `XRP_MAINNET_ENABLED` defaults to enabled and is independent of whether an endpoint exists. With none configured, `XRP_RPC_HTTP_URL_MAINNET` is `undefined` by design on `ic`/`beta` and `xrpHttpRpcUrl` throws — so every balance and history request fails and no send can start.

**Analysis**: I checked the deploy workflow rather than assuming "the secrets might be absent". `VITE_XRP_RPC_URL_MAINNET` is wired for **staging** and **beta** only. There is no `ic` or prod arm at all — zero matches for one. So beta depends on whether its secret is actually provisioned, and production cannot have an endpoint no matter what is provisioned, because nothing passes one to the build.

That makes this not a race between two rollout steps but a guaranteed outcome for `ic` as the PR stands: XRP appears in the network list, the worker starts, and every request throws.

**Severity**: High, as filed. A chain that is visible, selectable and wholly non-functional in production.

**Fix**: Two ways, and they are a sequence rather than alternatives.

1. The suggested changeset — enablement requires a non-empty endpoint on `PROD`/`BETA`. It makes the failure mode "XRP is absent" instead of "XRP is broken", which is the right way round, and it is self-correcting: the moment the secret is wired, XRP appears.
2. Wire `VITE_XRP_RPC_URL_MAINNET` for `ic` in `deploy-to-environment.yml` and provision the secret. The second half is a repository secret and not something this PR can carry, so the PR cannot make XRP work in production on its own.

Taking 1 now regardless of 2, because it is the guard that stops this recurring: without it, any future environment enabled before it is wired ships the same broken state silently.

🤖 Claude Opus 5

File: src/frontend/src/lib/workers/exchange.worker.ts [hunk 2]
```diff
@@ -372,6 +383,7 @@ const fetchProviderFallbackPrices = async ({
 	const fillBtc = COINGECKO_FALLBACK_PROVIDER_ENABLED && missingBtc;
 	const fillIcp = COINGECKO_FALLBACK_PROVIDER_ENABLED && missingIcp;
 	const fillSol = COINGECKO_FALLBACK_PROVIDER_ENABLED && missingSol;
+	const fillXrp = COINGECKO_FALLBACK_PROVIDER_ENABLED && missingXrp;
```
[the remark this answers is not in the document]
Comment (sbpublic) [reply src/frontend/src/lib/workers/exchange.worker.ts:386]: Correct.

**Summary**: The predicate lists `!fillEth && !fillBtc && !fillIcp && !fillSol && !fillBnb && !fillPol`. `!fillXrp` is not there, though `fillXrp` is computed four lines above exactly like its siblings.

**Analysis**: It only bites when XRP is the *only* native the backend omitted — any other missing price makes the guard false and the fallback runs, XRP included. That is why the XRP pricing tests in this PR pass: they exercise the fallback, not the early return that skips it.

**Severity**: Medium. Not a wrong price, an absent one, and only in the single-token case — but that case is the likeliest one for a newly added chain, since XRP is the price the backend is most likely to be missing on its own.

**Fix**: Add `!fillXrp` to the guard, with a case where XRP is the only missing native so the early return is the thing under test rather than the fallback.

🤖 Claude Opus 5

File: docs/ai/PRODUCT.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ This document is the living description of OISY's product behaviors. It is read
 
 ## What is OISY
 
-OISY is a browser-based, network-custodial, multi-chain wallet powered by the Internet Computer's chain fusion technology. It lets users receive, hold, and send native ICP, ICRC-1, ETH, ERC-20, and BTC without browser extensions or mobile apps. Keys are never held by a single entity — they are generated and managed using threshold ECDSA across ICP replica nodes.
+OISY is a browser-based, network-custodial, multi-chain wallet powered by the Internet Computer's chain fusion technology. It lets users receive, hold, and send native ICP, ICRC-1, ETH, ERC-20, BTC, SOL, and XRP without browser extensions or mobile apps. Keys are never held by a single entity — they are generated and managed using threshold ECDSA across ICP replica nodes.
```
[the remark this answers is not in the document]
Comment (sbpublic) [reply docs/ai/PRODUCT.md]: Correct.

**Summary**: The sentence lists SOL and XRP among the supported assets and says keys "are generated and managed using threshold ECDSA". Both sign with threshold Schnorr over Ed25519, and the XRP section added in this PR says Ed25519 outright.

**Analysis**: The list grew and the clause after it did not. It is the document's opening description, so it is the sentence most likely to be quoted elsewhere, and it now contradicts a section of the same file.

**Severity**: Low. Documentation, though a factual claim about how keys work.

**Fix**: Taking the suggested changeset — "threshold cryptography" is accurate for all of them, and the per-chain sections already carry the specifics for anyone who needs them.

🤖 Claude Opus 5

File: src/frontend/src/env/networks/networks.xrp.env.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sbpublic) [reply src/frontend/src/env/networks/networks.xrp.env.ts:15]: Correcting this reply — the reasoning in it was wrong, and the change it describes is reverted in 089f15195.

**What I got wrong**: I wrote that `deploy-to-environment.yml` wires `VITE_XRP_RPC_URL_MAINNET` for staging and beta only, "there is no `ic` arm", and concluded production could not have an endpoint. True but meaningless. That workflow has no `ic` target at all — its `network` choices are staging, beta, `test_fe_1-6`, `test_be_1` and audit — and no variable has an `ic` arm, Alchemy, Infura, QuickNode, Etherscan and CoinGecko included. Production is not deployed from this repo's workflows, so the absence proves nothing about XRP, and the endpoint's wiring there is exactly as complete as every other provider's.

**Also wrong in kind**: gating a chain's visibility on a credential being present is a mechanism no other chain has. ETH and SOL ship enabled and let their requests fail when a key is missing.

**What replaced it** (be526d9d3): the endpoint moves to `env/rest/xrpl.env.ts`, where every other provider credential lives, and the `xrplcluster.com` fallback is dropped — so a build without the variable has no XRPL endpoint and `xrpHttpRpcUrl` throws, exactly as a build without an Alchemy key fails its requests. The variable keeps its name and value shape, so no workflow or secret changes.

The finding itself was still worth raising; the fix it prompted was the wrong shape.

🤖 Claude Opus 5

File: src/frontend/src/env/networks/networks.xrp.env.ts [hunk 4]
```diff
@@ -11,43 +10,9 @@ import type { XrpNetwork } from '$xrp/types/network';
...
- * provider from a forgotten mock, which is the one thing this is here to prevent. Any spec that
- * needs an endpoint mocks the module.
- */
-export const XRP_RPC_HTTP_URL_MAINNET = TEST
-	? undefined
-	: (import.meta.env.VITE_XRP_RPC_URL_MAINNET ??
-		(PROD || BETA ? undefined : 'https://xrplcluster.com'));
+export const XRP_MAINNET_ENABLED = parseEnabledMainnetBoolEnvVar(
+	import.meta.env.VITE_XRP_MAINNET_DISABLED
+);
```
[the remark this answers is not in the document]
Comment (sbpublic) [reply src/frontend/src/env/networks/networks.xrp.env.ts:15]: Correct, and it is not specific to a missing endpoint — any sustained RPC failure produces it.

**Summary**: `resetWallet` leaves the XRP transactions entry `undefined`, `syncWalletError` never writes it, and `isTransactionsStoreInitialized` counts only `!== undefined` — so once XRP joins `transactionsStoreWithTokens`, one token that never loaded holds the aggregate gate open for good.

**Analysis**: every link checked at `1a8194104`. `resetWallet` calls `xrpTransactionsStore.clear(tokenId)` (`xrp-listener.services.ts:63`) and runs on every worker start and address change (`worker.xrp-wallet.services.ts:116,146`). The scheduler retries the balance ten times, clears its own store and posts the error; the listener's `syncWalletError` then resets `balancesStore` alone. That is deliberate — it was changed in an earlier round of #13595 so a transient failure would not discard history the scheduler cannot refetch, since it passes no `marker` and only ever re-reads the newest page. The cost was not seen: the entry stays `undefined`, which is exactly what the gate reads.

XRP is the only chain that does this. BTC (`btc-listener.services.ts:70`) and SOL (`sol-listener.services.ts:132`) call `reset`, IC (`ic-listener.services.ts:38`) calls `nullify`; all three write `null`, which counts as initialized and settles the gate while discarding rows.

Two consequences rather than one, and one correction to the finding:

- Skeletons forever, but only when the account has no transactions on any chain — `areTransactionsStoresLoading` is `(someNullish || someNotInitialized) && allEmpty`, so a populated account shows its rows and just never leaves the loading state internally.
- Levelling never runs. `allStoresAreLoaded` gates the `$effect` at `AllTransactionsLoader.svelte:238`, so `levelNewcomers` never fires and tokens are never pulled to a common floor — a chain that returned an old row is displayed while a busier one still holds only its newest page, and scrolling reveals in-memory rows without asking for more, so the gap stays. This is not "Activity pagination disabled": `loadMore` is unaffected.

**Severity**: High is right. It needs no misconfiguration — a provider outage or a rate limit on first load is enough, and it does not recover until a sync succeeds.

**Fix**: settle the entry in `syncWalletError`, but only when nothing has ever loaded:

```ts
if (isNullish(get(xrpTransactionsStore)?.[tokenId])) {
	xrpTransactionsStore.nullify(tokenId);
}
```

That meets both halves of the constraint: a token whose first load never succeeded stops holding the gate, and an entry that already holds rows keeps them — which is why it is not the unconditional `reset` the other three chains do. Excluding a failed worker from the aggregate gate would need failure state plumbed into `transactionsStoreWithTokens` for one chain's benefit; this is the same outcome in the store the gate already reads.

🤖 Claude Opus 5

File: src/frontend/src/xrp/services/xrp-listener.services.ts [hunk 5]
```diff
@@ -83,6 +83,20 @@ export const syncWalletError = ({
...
+	// is not `undefined` — so a token whose FIRST load never succeeded holds the aggregate Activity
+	// gate open for good: an otherwise-empty account stays on skeletons, and `levelNewcomers` never
+	// runs, leaving the chains at uneven floors. A provider outage or a rate limit on first load is
+	// enough; no misconfiguration needed.
+	//
+	// Written only in that never-loaded case, which is what separates this from the unconditional
+	// `reset`/`nullify` the other chains do: an entry that already holds rows keeps them, for the
+	// reason above.
+	if (isNullish(get(xrpTransactionsStore)?.[tokenId])) {
+		xrpTransactionsStore.nullify(tokenId);
```
[the remark this answers is not in the document]
Comment (sbpublic) [reply src/frontend/src/xrp/services/xrp-listener.services.ts:111]: Correct. The fix settled the wrong half of the problem.

**Summary**: an `account_tx` failure never reaches `syncWalletError`. `xrp-wallet.scheduler.ts:173-176` folds the rejection into `undefined`, `syncWalletData` then omits `newTransactions` from the post-message, and `syncWallet` returns before touching the store — so with a healthy balance and a failing history the entry stays `undefined` and the gate stays open, silently, with no toast.

**Analysis**: verified at `c272839a7`. `loadAndSyncBalance` awaits the transactions promise before calling `syncWalletData` (`:156`), so by then the history has resolved or failed; absent `newTransactions` means failed, never "still loading". The early return at `:41` is deliberate — an empty array would claim the account has no history on the strength of a request that failed — and my last commit only covered the path where the *balance* fails too.

It isn't hypothetical on this integration. `account_tx` is the heavier call and is rate-limited before `account_info`; and the configured provider is a Clio endpoint with a method whitelist, so a method not on it returns 401 `-32611` on every history read while balances keep working.

**Severity**: Medium is right. It needs history to fail while the balance succeeds, which is narrower than the previous one, but it is also quieter — nothing is reported at all.

**Fix**: the same conditional write, on the omitted-history path:

```ts
if (isNullish(newTransactions)) {
	if (isNullish(get(xrpTransactionsStore)?.[tokenId])) {
		xrpTransactionsStore.nullify(tokenId);
	}
	return;
}
```

`null` is not the empty page the early return exists to prevent — it is "asked, nothing to show", which is what the ledger read actually established. Rows that already loaded are untouched, and the per-token XRP view keeps its skeleton because `xrpTransactionsInitialized` uses `nonNullish`.

**One cost, stated rather than hidden**: once settled, the aggregate view reports XRP as having no activity instead of spinning. That is better than a permanent skeleton, but it is still wrong in a way the user cannot see. Surfacing an explicit history-failure state — your second suggestion — is the real answer and wants its own change; it needs a new field on the post-message and a decision about how a partial outage is presented.

🤖 Claude Opus 5

**Comment by sbpublic** [comment]
**Previously missed: Document total reserve including owner reserves** — correct.

**Summary**: the "Base reserve" section says the maximum sendable subtracts the base reserve. The send path subtracts the base reserve *plus* `ownerCount × XRP_OWNER_RESERVE_DROPS`, so any account holding a trust line, offer or escrow retains more than the document describes.

**Analysis**: `getXrpReserveDrops` (`xrp-send.utils.ts:15`) is `XRP_BASE_RESERVE_DROPS + BigInt(ownerCount) * XRP_OWNER_RESERVE_DROPS`, and it is what the UI shows — `XrpFeeContext.svelte:94` loads `ownerCount` from `account_info` and feeds it to `reserveStore`. The code was right; only the product description lagged, from when the base reserve was all this handled.

Worth being accurate about rather than deleting, because it is the difference a user sees: an account with a few trust lines keeps noticeably more than the section implies, and the gap is exactly what makes a max-send fail if we describe it wrongly.

**Severity**: Low is right. Documentation only, no behaviour affected.

**Fix**: rewriting the section as the total account reserve — base plus one owner reserve per owned ledger object — and renaming it accordingly. No figures, matching the rest of `PRODUCT.md` and the fact that both values are validator-configurable (`xrp.constants.ts` carries a TODO to read them from `server_state`).

🤖 Claude Opus 5

File: docs/ai/spec-driven-development/specs/2026-07-24-feat-xrp-ledger-integration.md [hunk 6]
```diff
@@ -386,17 +386,18 @@ exposing a half-built chain to users.
...
+| 10    | `feat/frontend/xrp-enable`              | Phase 9    | **This PR.** Remove the temporary `XRP_MAINNET_DISABLED_OVERRIDE` (so `VITE_XRP_MAINNET_DISABLED` governs XRP like every other chain), map XRP in `networkIdToKey` / `keyToNetworkId`, exchange-rate/price lookup, `PRODUCT.md` + `docs/ai/integrations/xrpl.md` updates, and the suite-wide network/token expectations that only change once XRP is enabled.                                                                                                                                                                                                                 |
```
[the remark this answers is not in the document]
Comment (sbpublic) [reply docs/ai/spec-driven-development/specs/2026-07-24-feat-xrp-ledger-integration.md:408]: Correct, with one qualification.

**Summary**: lines 372-384 describe the pre-enable state in the present tense — `XRP_MAINNET_DISABLED_OVERRIDE` "force-disables" (`:375`), `SUPPORTED_XRP_NETWORKS`/`SUPPORTED_XRP_TOKENS` "resolve to empty arrays" (`:379`). This phase removes the override, so the canonical spec now describes a state that no longer exists.

**Qualification**: "no CI/deploy env plumbing" (`:382`) is true of what it is about — enablement is still `VITE_XRP_MAINNET_DISABLED` alone, entirely in code, and this PR reverted a gate that would have broken that. It is misleading only because the paragraph reads as if the integration needs none at all, and the endpoint does: `VITE_XRP_RPC_URL_MAINNET_STAGING`/`_BETA`, forwarded by `deploy-to-environment.yml`, plus a production value injected by the build. That plumbing arrived in Phase 3 with the RPC client, not here.

**Severity**: Low is right. Documentation of a shipped state.

**Fix**: putting the override paragraph in the past tense and adding a sentence for the shipped state, rather than rewriting it away — a spec is a snapshot of a decision and the rollout it describes actually happened. The endpoint-plumbing exception goes in the same sentence, since the "entirely in code" claim is what reads wrong.

🤖 Claude Opus 5

**Comment by sbpublic** [comment]
**Previously missed: Align XRP configuration guidance with populated local endpoint** — correct, and self-inflicted: I populated `.env.example` after writing those three comments.

**Summary**: `xrpl.env.ts:28-29`, `xrpl.env.spec.ts:44-45` and `xrpl.md:285-287` all give "a `.env` copied from `.env.example`" as the everyday way an empty value arrives. It now yields `https://xrplcluster.com`, so that example is false.

**Analysis**: only the example is wrong, not the behaviour. An empty value still needs handling — a secret created blank, an `env-override` line with no value, a developer who clears the line — and the reasoning holds: `??` would keep `''`, `xrpHttpRpcUrl` would pass its check and resolve every request against the app's own origin. So the test stays; the justification loses one of its two cases.

**Severity**: Low is right. Comments and docs.

**Fix**: dropping the copied-example clause in all three, keeping the uncreated-secret case.

🤖 Claude Opus 5
